datafusion-cli/src/functions.rs

```diff
@@ -0,0 +1,192 @@
+    [ LATERAL ] ( select ) [ AS ] alias [ ( column_alias [, ...] ) ]
+    with_query_name [ [ AS ] alias [ ( column_alias [, ...] ) ] ]
+    [ LATERAL ] function_name ( [ argument [, ...] ] )
+                [ WITH ORDINALITY ] [ [ AS ] alias [ ( column_alias [, ...] ) ] ]
+    [ LATERAL ] function_name ( [ argument [, ...] ] ) [ AS ] alias ( column_definition [, ...] )
+    [ LATERAL ] function_name ( [ argument [, ...] ] ) AS ( column_definition [, ...] )
+    [ LATERAL ] ROWS FROM( function_name ( [ argument [, ...] ] ) [ AS ( column_definition [, ...] ) ] [, ...] )
+                [ WITH ORDINALITY ] [ [ AS ] alias [ ( column_alias [, ...] ) ] ]
+    from_item [ NATURAL ] join_type from_item [ ON join_condition | USING ( join_column [, ...] ) [ AS join_using_alias ] ]
+
+and grouping_element can be one of:
+
+    ( )
+    expression
+    ( expression [, ...] )
+
+and with_query is:
+
+    with_query_name [ ( column_name [, ...] ) ] AS [ [ NOT ] MATERIALIZED ] ( select | values | insert | update | delete )
+
+TABLE [ ONLY ] table_name [ * ]";
+                println!("{}", details)
+            }
+            Function::Explain => {
+                let details = "
+Command:     EXPLAIN
+Description: show the execution plan of a statement
+Syntax:
+EXPLAIN [ ANALYZE ] statement
+";
+                println!("{}", details)
+            }
+            Function::Show => {
+                let details = "
+Command:     SHOW
+Description: show the value of a run-time parameter
+Syntax:
+SHOW name
+";
+                println!("{}", details)
+            }
+            Function::CreateTable => {
+                let details = "
+Command:     CREATE TABLE
+Description: define a new table
+Syntax:
+CREATE [ EXTERNAL ]  TABLE table_name ( [
+  { column_name data_type }
+    [, ... ]
+] )
+";
+                println!("{}", details)
+            }
+            Function::CreateTableAs => {
+                let details = "
+Command:     CREATE TABLE AS
+Description: define a new table from the results of a query
+Syntax:
+CREATE TABLE table_name
+    [ (column_name [, ...] ) ]
+    AS query
+    [ WITH [ NO ] DATA ]
+";
+                println!("{}", details)
+            }
+            Function::Insert => {
+                let details = "
+Command:     INSERT
+Description: create new rows in a table
+Syntax:
+INSERT INTO table_name [ ( column_name [, ...] ) ]
+    { VALUES ( { expression } [, ...] ) [, ...] }
+";
+                println!("{}", details)
+            }
+            Function::DropTable => {
+                let details = "
+Command:     DROP TABLE
+Description: remove a table
+Syntax:
+DROP TABLE [ IF EXISTS ] name [, ...]
+";
+                println!("{}", details)
+            }
+        }
+        Ok(())
+    }
+}
+
+impl FromStr for Function {
+    type Err = ();
+
+    fn from_str(s: &str) -> std::result::Result<Self, Self::Err> {
+        Ok(match s.to_uppercase().as_str() {
```

Comment:
The cli crashes when function name surrounded whitespace  so how about:
```suggestion
        Ok(match s.trim().to_uppercase().as_str() {
```